newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+    const { event } = this.props;
+    const eventType = event.getType();
+    let textColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      textColors = {
+        red: commentEvent.getTextColorRed(),
+        green: commentEvent.getTextColorGreen(),
+        blue: commentEvent.getTextColorBlue(),
+      };
+      return textColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      var groupEvent = gd.asGroupEvent(event);
+      const r = groupEvent.getBackgroundColorR(),
+        g = groupEvent.getBackgroundColorG(),
+        b = groupEvent.getBackgroundColorB();
+
+      textColors = (r + g + b) / 3 > 200 ? 'black' : 'white'; //Because text color is not supported by Core
+      return textColors;
+    } else {
+      console.error('Dialog was opened for an unsupported event text color.');
+      return '';
+    }
+  };
+
+  getBackgroundColors = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    let backgroundColors;
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      const commentEvent = gd.asCommentEvent(event);
+      backgroundColors = {
+        red: commentEvent.getBackgroundColorRed(),
+        green: commentEvent.getBackgroundColorGreen(),
+        blue: commentEvent.getBackgroundColorBlue(),
+      };
+
+      return backgroundColors;
+    } else if (eventType === 'BuiltinCommonInstructions::Group') {
+      const groupEvent = gd.asGroupEvent(event);
+      backgroundColors = {
+        red: groupEvent.getBackgroundColorR(),
+        green: groupEvent.getBackgroundColorG(),
+        blue: groupEvent.getBackgroundColorB(),
+      };
+
+      return backgroundColors;
+    } else {
+      console.error(
+        'Dialog was opened for an unsupported event background color'
+      );
+      return '';
+    }
+  };
+
+  state = {
+    textValue: this.getComment(),
+    textColors: this.getTextColors(),
+    backgroundColors: this.getBackgroundColors(),
+  };
+
+  render() {
+    const { event, onApply, onClose } = this.props;
+    const eventType = event.getType();
+
+    return (
+      <Dialog
+        title={<Trans>Text editor</Trans>}
+        onRequestClose={onClose}
+        open
+        noMargin
+        actions={[
+          <FlatButton
+            key="close"
+            label={<Trans>Cancel</Trans>}
+            primary={false}
+            onClick={onClose}
+          />,
+          <FlatButton
+            key={'Apply'}
+            label={<Trans>Apply</Trans>}
+            primary
+            keyboardFocused
+            onClick={() => {
+              if (eventType === 'BuiltinCommonInstructions::Comment') {
+                this.setComment();
+              } else if (eventType === 'BuiltinCommonInstructions::Group') {
+                this.setName();
+              }
+              this.setTextColors();
+              this.setBackgroundColors();
```

Comment:
See my previous comment, ideally we do this as a single operation in a function, could make things a bit clearer.